# HG changeset patch
# User Rik <rik@octave.org>
# Date 1386092154 28800
#      Tue Dec 03 09:35:54 2013 -0800
# Node ID 70e83c641b48177c7e83b03a6515cc687c676e28
# Parent  7c9148b235ebf52a6de9f8f1c2b5b4d6aef0f427
Accept a bare directory name for rundemos and runtests.

* rundemos.m, runtests.m: Use isdir to check whether input is a directory.

diff --git a/scripts/testfun/rundemos.m b/scripts/testfun/rundemos.m
--- a/scripts/testfun/rundemos.m
+++ b/scripts/testfun/rundemos.m
@@ -33,28 +33,27 @@
 function rundemos (directory)
 
   if (nargin == 0)
     dirs = ostrsplit (path (), pathsep ());
     do_class_dirs = true;
   elseif (nargin == 1)
     if (is_absolute_filename (directory))
       dirs = {directory};
-    elseif (is_rooted_relative_filename (directory))
+    elseif (isdir (directory))
       dirs = {canonicalize_file_name(directory)};
     else
       if (directory(end) == filesep ())
         directory = directory(1:end-1);
       endif
       fullname = find_dir_in_path (directory);
-      if (! isempty (fullname))
-        dirs = {fullname};
-      else
+      if (isempty (fullname))
         error ("rundemos: DIRECTORY argument must be a valid pathname");
       endif
+      dirs = {fullname};
     endif
     do_class_dirs = false;
   else
     print_usage ();
   endif
 
   for i = 1:numel (dirs)
     d = dirs{i};
diff --git a/scripts/testfun/runtests.m b/scripts/testfun/runtests.m
--- a/scripts/testfun/runtests.m
+++ b/scripts/testfun/runtests.m
@@ -33,28 +33,27 @@
 function runtests (directory)
 
   if (nargin == 0)
     dirs = ostrsplit (path (), pathsep ());
     do_class_dirs = true;
   elseif (nargin == 1)
     if (is_absolute_filename (directory))
       dirs = {directory};
-    elseif (is_rooted_relative_filename (directory))
+    elseif (isdir (directory))
       dirs = {canonicalize_file_name(directory)};
     else
       if (directory(end) == filesep ())
         directory = directory(1:end-1);
       endif
       fullname = find_dir_in_path (directory);
-      if (! isempty (fullname))
-        dirs = {fullname};
-      else
+      if (isempty (fullname))
         error ("runtests: DIRECTORY argument must be a valid pathname");
       endif
+      dirs = {fullname};
     endif
     do_class_dirs = false;
   else
     print_usage ();
   endif
 
   for i = 1:numel (dirs)
     d = dirs{i};
